### PR TITLE
Bug 227

### PR DESCRIPTION
**Pull request by feng-tao**
Fix available for bug 227(https://github.com/linkedin/naarad/issues/277)

### CODE REVIEW

**Comment by richardhsu** [comment]
So this fix will have GLOBAL section always overriding any subsections. What do others think?

Should GLOBAL be a full override or only if settings are not present in other sections?

**Comment by RiteshMaheshwari** [comment]
We should not have GLOBAL override subsection values. GLOBAL is equivalent to default.

**Comment by zhenyun** [comment]
right, should be the opposite:   Sub-section overrides global.

________________________________

**Comment by feng-tao** [comment]
Fix the issue mentioned by the discussion. Now global section only takes effect if no subsection specified.

File: src/naarad/__init__.py [hunk 1]
```diff
@@ -426,10 +427,16 @@ def _process_naarad_config(self, config, analysis):
             naarad.utils.parse_basic_metric_options(config, section)
           sar_metrics = naarad.utils.get_all_sar_objects(metrics, infile, hostname, output_directory, label, ts_start,
                                                          ts_end, None)
+          if ts_start is not None and sar_metric.ts_start is None and ts_end is not None and sar_metric.ts_end is None:
```
Comment (RiteshMaheshwari) [inline src/naarad/__init__.py]: @feng-tao , might be better to separate out the ts_start and ts_end conditions. Users should be able to specify only ts_start or ts_end and still have that take effect.

**Comment by feng-tao** [comment]
@RiteshMaheshwari  & @zhenyun  Thanks for the comment. I just push another fix. So now if subsection specifies ts_start, it will pick up the global ts_end, only if global's ts_end is larger than self's ts_start. And if subsection specifies ts_end, it will pick the global ts_start, only if global's ts_start is smaller than self's ts_end.

File: src/naarad/__init__.py [hunk 2]
```diff
@@ -426,10 +427,40 @@ def _process_naarad_config(self, config, analysis):
             naarad.utils.parse_basic_metric_options(config, section)
           sar_metrics = naarad.utils.get_all_sar_objects(metrics, infile, hostname, output_directory, label, ts_start,
                                                          ts_end, None)
+          if ts_start is not None:
+            if sar_metric.ts_start is None:
+              if sar_metric.ts_end is None:
```
Comment (RiteshMaheshwari) [inline src/naarad/__init__.py]: Just wondering if these 5 lines combined into one if statement:

```
if sar_metric.ts_end is None or sar_metric.ts_end > ts_start:
  sar_metric.ts_start = ts_start
```

Same below.

**Comment by feng-tao** [comment]
This wont check whether sar_metric.ts_start has been set or not....It will get overrided by ts_start if the condition match.

**Comment by feng-tao** [comment]
Modify the code and combine it into two lines...

**Comment by richardhsu** [comment]
I think it's ok to have the two bug fixes merged in #310 together, but since they are separate issues it would be nicer to have them separate so they can be reviewed in isolation so for future issues please have separate pull requests and fixes. But we can review #310 and get that going for both bug issues. Thanks! Going to close this one.

**Pull request closed without merging**